src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/BlobUtilities.cs

```diff
@@ -62,60 +54,27 @@ public static void WriteByte(this byte[] buffer, int start, byte value)
 
         public static void WriteUInt16(this byte[] buffer, int start, ushort value)
         {
-            fixed (byte* ptr = &buffer[start])
-            {
-                unchecked
-                {
-                    ptr[0] = (byte)value;
-                    ptr[1] = (byte)(value >> 8);
-                }
-            }
+            BinaryPrimitives.WriteUInt16LittleEndian(buffer.AsSpan(start), value);
```

Comment:
Doesn't this negatively impact the performance of this method?  Have you validated there's no negative impact on consuming performance of this and the other Write methods here?

Reply:
I changed it to use `Unsafe.WriteUnaligned`, removing the check that the array has enough elements, introduced by `BinaryPrimitives`.